src/client/features/enrichment/cy/cluster-label-categorization/index.js

```diff
@@ -0,0 +1,72 @@
+const _ = require('lodash');
+
+
+const { stemmer } = require('porter-stemmer');
+const stopWords = require('./stopwords.json');
+const flaggedWords = require('./flaggedwords.json');
+
+let textCharacterization = (text) => {
+  let delimiterRegex = /[\t \n\r\f!\\#$%&()*+,.<=>?@[^\\\]`_{|}~\\'"]/;
+  let filterWords = new Set([...stopWords, flaggedWords]);
+  let wordPairMap = new Map();
+  let wordOccurenceMap = new Map();
+  let stemToWordsMap = new Map();
+
+  let incrKey = ( m, k ) => {
+    if( m.has( k ) ){
+      m.set(k, m.get(k) + 1);
+    } else {
+      m.set( k, 1 );
+    }
+  };
+  let appendKey = ( m, k, v ) => {
+    if( m.has( k ) ){
+      m.set(k, m.get(k).add(v) );
+    } else {
+      m.set( k, new Set([v]) );
+    }
+  };
+
+  let wordStems = _.uniq(text.split(delimiterRegex).filter( word => !filterWords.has(word) && word !== '' ).map( word => word.toLowerCase() ).map( word => {
+    incrKey( wordOccurenceMap, word );
+    let wordStem = stemmer(word);
+    appendKey( stemToWordsMap, wordStem, word );
+    return wordStem;
+   } ) );
+
+
+  for( let i = 0; i < wordStems.length - 1; i++ ){
```

Comment:
Does this block ever get used?

Reply:
This has been removed for now, but it was part of something that the original word cloud app did.  I wasn't able to port it fully but it did something like this:

1.  compute the occurrences of each pair of words (A,B)
2.  compute probability (P(B|A)P(A))/(P(A)P(B))
3. remove pairs that have probability < cutOff ( I don't know the value of cutOff )
4. perform a merge operation on some other set that I don't know all the details about

Since the results seem okay without these things, I think I can remove it for now.  It will take a bit of time for me to understand the original code better.